pkg/datagatherer/k8s/dynamic.go

```diff
@@ -79,6 +83,16 @@ func (c *ConfigDynamic) validate() error {
 		errors = append(errors, "invalid configuration: GroupVersionResource.Resource cannot be empty")
 	}
 
+	for i, selectorString := range c.FieldSelectors {
+		if selectorString == "" {
+			errors = append(errors, fmt.Sprintf("invalid field selector %d: must not be empty", i))
+		}
+		_, err := fields.ParseSelector(selectorString)
+		if err != nil {
+			errors = append(errors, fmt.Sprintf("invalid field selector %d: %s", i, err))
+		}
+	}
```

Comment:
The selectors are validated here but the validation is not very good.
ParseSelector will happily parse strings like "type != foo" into 
```go
fields.Requirement {
  Op: "!=",
  Field: "type " // with trailing space
  Value: " foo" // with leading space
}
```

And then later the client-go Reflector will fail with an error like "unknown field: type " but you won't be able to see the trailing space, because it's not quoted.